### PR TITLE
token implementation fixes

### PR DESCRIPTION
**Pull request by sup25**
- Implemented the blacklist functionality correctly by generating both the accessToken and refreshToken, followed by querying the database to verify the blacklist status of the token.
- Verified the token blacklist status by querying the database immediately after token generation to ensure proper handling of blacklisted tokens.

Reviewer 
@Elnatanv

### CODE REVIEW

File: app/api/v1/user/auth/refreshToken/route.js [hunk 1]
```diff
@@ -23,6 +23,19 @@ export async function POST(request) {
     );
   } catch (error) {
     console.error("Error refreshing token:", error.message);
+    try {
```
Comment (Elnatanv) [inline app/api/v1/user/auth/refreshToken/route.js]: @sup25 This should be a service: invalidateUserTokens . pass the token as argument.  await invalidateUserTokens(refreshToken)

File: app/api/v1/user/auth/tokenService/generateTokens/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  const decodedAccessToken = jwt.decode(accessToken);
+
+  const expDate = new Date(decodedAccessToken.exp * 1000).toISOString();
+
+  await prisma.token.create({
+    data: {
+      token: accessToken,
+      userId: user.id,
+      type: "ACCESS_TOKEN ",
```
Comment (Elnatanv) [inline app/api/v1/user/auth/tokenService/generateTokens/index.js]: you have a space after ACCESS_TOKEN

File: app/api/v1/user/auth/tokenService/generateTokens/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      black_list: false,
+      expire: expDate,
+    },
+  });
+
+  return accessToken;
+};
+
+const generateRefreshToken = async (user) => {
+  await prisma.token.updateMany({
```
Comment (Elnatanv) [inline app/api/v1/user/auth/tokenService/generateTokens/index.js]: this is the same service as invalidateUserTokens . but you dont need to do it inside this service. its a business logic. we should do invalidate on a controller level, then to generate.

File: app/api/v1/user/auth/tokenService/generateTokens/index.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    },
+  });
+
+  const refreshToken = jwt.sign({ id: user.id }, config.refreshTokenSecret, {
+    expiresIn: config.refreshTokenExpiration,
+  });
+
+  const decodedRefreshToken = jwt.decode(refreshToken);
+
+  const expDate = new Date(decodedRefreshToken.exp * 1000).toISOString();
```
Comment (Elnatanv) [inline app/api/v1/user/auth/tokenService/generateTokens/index.js]: should be a util function - takes decodedRefreshToken.exp as argument. reutrns the date.

File: app/api/v1/user/auth/tokenService/verifyToken/index.js [hunk 5]
```diff
@@ -9,6 +12,18 @@ const verifyToken = async (token, isRefreshToken = false) => {
 
     const payload = jwt.verify(token, verifySecretKey);
 
+    const blacklistedToken = await prisma.token.findFirst({
```
Comment (Elnatanv) [inline app/api/v1/user/auth/tokenService/verifyToken/index.js]: create a service check if black listed. put the logic there. and call it inside auth wrapper function. before verifying the token, means if token is black listed (or not exist) - we want to return error 401 . and exist. inside where:{} remove id. we dont care about the id.

**Review by Elnatanv — CHANGES_REQUESTED** [review]

File: app/api/v1/user/auth/refreshToken/route.js [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
+import verifyToken from "../tokenService/verifyToken";
+import { generateAccessToken } from "../tokenService/generateTokens";
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
 
 export async function POST(request) {
   const { refreshToken } = await request.json();
   console.log("refresh token: ", refreshToken);
   try {
```
Comment (Elnatanv) [inline app/api/v1/user/auth/refreshToken/route.js]: All this try catch should be a controller. refreshToken controller

**Review by Elnatanv — CHANGES_REQUESTED** [review]

File: app/api/v1/user/auth/refreshToken/route.js (hunk 1 above)
Comment (sup25) [reply app/api/v1/user/auth/refreshToken/route.js]: Done

File: app/api/v1/user/auth/tokenService/generateTokens/index.js (hunk 2 above)
Comment (sup25) [reply app/api/v1/user/auth/tokenService/generateTokens/index.js]: Done

File: app/api/v1/user/auth/tokenService/generateTokens/index.js (hunk 4 above)
Comment (sup25) [reply app/api/v1/user/auth/tokenService/generateTokens/index.js]: Done

File: app/api/v1/user/auth/tokenService/generateTokens/index.js (hunk 3 above)
Comment (sup25) [reply app/api/v1/user/auth/tokenService/generateTokens/index.js]: Done

File: app/api/v1/user/auth/tokenService/verifyToken/index.js (hunk 5 above)
Comment (sup25) [reply app/api/v1/user/auth/tokenService/verifyToken/index.js]: Need to review

File: app/api/v1/user/auth/refreshToken/route.js (hunk 6 above)
Comment (sup25) [reply app/api/v1/user/auth/refreshToken/route.js]: Done

**Pull request merged**